product_brand/product_brand.py

```diff
@@ -0,0 +1,301 @@
+    return pricelist
+
+
+class table_compute(object):
+    def __init__(self):
+        self.table = {}
+
+    def _check_place(self, posx, posy, sizex, sizey):
+        res = True
+        for y in range(sizey):
+            for x in range(sizex):
+                if posx+x >= PPR:
+                    res = False
+                    break
+                row = self.table.setdefault(posy+y, {})
+                if row.setdefault(posx+x) is not None:
+                    res = False
+                    break
+            for x in range(PPR):
+                self.table[posy+y].setdefault(x, None)
+        return res
+
+    def process(self, products):
+        # Compute products positions on the grid
+        minpos = 0
+        index = 0
+        maxy = 0
+        for p in products:
+            x = min(max(p.website_size_x, 1), PPR)
+            y = min(max(p.website_size_y, 1), PPR)
+            if index >= PPG:
+                x = y = 1
+
+            pos = minpos
+            while not self._check_place(pos % PPR, pos/PPR, x, y):
+                pos += 1
+            # if 21st products (index 20) and the last line is full (PPR products in it), break
+            # (pos + 1.0) / PPR is the line where the product would be inserted
+            # maxy is the number of existing lines
+            # + 1.0 is because pos begins at 0, thus pos 20 is actually the 21st block
+            # and to force python to not round the division operation
+            if index >= PPG and ((pos + 1.0) / PPR) > maxy:
+                break
+
+            if x == 1 and y == 1:   # simple heuristic for CPU optimization
+                minpos = pos/PPR
+
+            for y2 in range(y):
+                for x2 in range(x):
+                    self.table[(pos/PPR)+y2][(pos % PPR)+x2] = False
+            self.table[pos/PPR][pos % PPR] = {
+                'product': p, 'x': x, 'y': y,
+                'class': " ".join(map(lambda x: x.html_class or '',
+                                      p.website_style_ids))
+            }
+            if index <= PPG:
+                maxy = max(maxy, y+(pos/PPR))
+            index += 1
+
+        # Format table according to HTML needs
+        rows = self.table.items()
+        rows.sort()
+        rows = map(lambda x: x[1], rows)
+        for col in range(len(rows)):
+            cols = rows[col].items()
+            cols.sort()
+            x += len(cols)
+            rows[col] = [c for c in map(lambda x: x[1], cols) if c != False]
+        return rows
+
+
+class website_sale(http.Controller):
+
+    def get_pricelist(self):
+        return get_pricelist()
+
+    @http.route([
+        '/shop', '/shop/page/<int:page>',
+        '/shop/category/<model("product.public.category"):category>',
+        '/shop/category/<model("product.public.category"):category>/page/<int:page>'
+        ], type='http', auth="public", website=True)
+    def shop(self, page=0, category=None, search='', **post):
```

Comment:
The same  thing as V8 module, here you are breaking inheritance in all methods, please use super.